### ISSUE TITLE
Build ASPECT in 'release mode only' causes external plugins to not compile

### ISSUE THREAD

**Issue by gassmoeller** [geodynamics/aspect#6017]
Something is not right with our cmake setup for plugins and the interaction with the different build types for ASPECT.

If I build ASPECT in release mode only I can no longer compile external plugins that try to link to that ASPECT build folder. I see the following error message (path shortened for readability):
```
.../finite_strain.cc:21:10: fatal error: aspect/material_model/simple.h: No such file or directory
   21 | #include <aspect/material_model/simple.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/finite_strain.dir/build.make:76: CMakeFiles/finite_strain.dir/finite_strain.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/finite_strain.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 75%] Built target finite_strain.release
make: *** [Makefile:136: all] Error 2
```
If I set up ASPECT in Debug or DebugRelease mode (just running cmake is enough, actually building not required) everything works as expected.
 
For my problem: When I run `make -n` I see that the compile/include flags are not set (only showing relevant line):
```
/usr/bin/c++ -Dfinite_strain_EXPORTS  -fPIC -MD -MT CMakeFiles/finite_strain.dir/finite_strain.cc.o -MF CMakeFiles/finite_strain.dir/finite_strain.cc.o.d -o CMakeFiles/finite_strain.dir/finite_strain.cc.o -c /home/rene/work/models/tests/2024-timo-composition-particles-profiling/main/finite_strain.cc
```

I have been poking around in [here](https://github.com/geodynamics/aspect/blob/506096610a964b48df34ca49f67257089b84ea9a/cmake/AspectConfig.cmake.in#L57) and to me it looks like we create the release target of the plugin with some dependency on the regular target. But the regular target seems to be set up as well, but does not get assigned any flags.

I have found a workaround: If I run `make finite_strain.release` everything compiles, but `make`  creates the error above. That looks to me like the build is trying to build the target  `finite_strain` (which is set up in my CMakeLists) as well, but that is not set up correctly by the plugin macro.

Files to reproduce with an ASPECT build folder that is only set to RELEASE build type:
[CMakeLists.txt](https://github.com/user-attachments/files/16782450/CMakeLists.txt)
[finite_strain.cc.txt](https://github.com/user-attachments/files/16782452/finite_strain.cc.txt)

Do we need to modify the plugin macro or the CMakeLists.txt?

**Comment by tjhei** [comment]
ping @bangerth . I am sure you broke it. :-)

**Comment by bangerth** [comment]
I'm sure too. I will have to investigate. I think I will have to make the same kind of changes to the plugin cmake scripts as I made to the ASPECT cmake scripts themselves...

**Comment by gassmoeller** [comment]
Luckily it was an easy fix. See #6070.

**Comment by gassmoeller** [comment]
Fixed by #6070.
